client/my-sites/domains/domain-management/settings/types.tsx

```diff
@@ -0,0 +1,8 @@
+import { SiteData } from 'calypso/state/ui/selectors/site-data';
+
+export type SettingsPageProps = {
+	currentRoute: string;
+	selectedDomainName: string;
+	selectedSite: SiteData | null;
+	hasDomainOnlySite: boolean | null;
```

Comment:
Since we're sure that we'd be receiving a `selectedSite` in this case, and consequently a `hasDomainOnlySite` boolean, I'd change the types to match the actual passed values:

```suggestion
	selectedSite: SiteData;
	hasDomainOnlySite: boolean;
```

Reply:
I totally agree, it make sense, thanks @rafaelgalani!